src/saml2/assertion.py

```diff
@@ -454,6 +455,9 @@ def post_entity_categories(maps, sp_entity_id=None, mds=None, required=None):
                         else:
                             attrs = []
 
+                        if attrs and no_aggregation:
+                            # clear restrictions if the found category is a no aggregation category
+                            restrictions = {}
```

Comment:
so, when we say that the order of the no-aggregate ECs is important we also mean that they MUST be defined last for this to work properly; iow, the order of the all ECs is important (not just the no-aggreate ones). For example:

```py
RELEASE = {
    ESI: MYACADEMICID_ESI,
    # order is important for no aggregation categories as the last one wins if more than one is found
    PERSONALIZED: REFEDS_PERSONALIZED_ACCESS,
    PSEUDONYMOUS: REFEDS_PSEUDONYMOUS_ACCESS,
    ANONYMOUS: REFEDS_ANONYMOUS_ACCESS,
}
```

is different from

```py
RELEASE = {
    # order is important for no aggregation categories as the last one wins if more than one is found
    PERSONALIZED: REFEDS_PERSONALIZED_ACCESS,
    PSEUDONYMOUS: REFEDS_PSEUDONYMOUS_ACCESS,
    ANONYMOUS: REFEDS_ANONYMOUS_ACCESS,
    ESI: MYACADEMICID_ESI,
}
```

even though `ESI` is not an EC in the `NO_AGGREGATION` set.

The first one, will result correctly in releasing the attributes related to the anonymous EC. But the second will result in releasing the attributes related to the anonymous EC and the ESI.